lib/screens/cocktail/cocktail_page.dart

```diff
@@ -0,0 +1,253 @@
+                                style: const TextStyle(
+                                  fontWeight: FontWeight.w500,
+                                ),
+                              ),
+                            );
+
+                            if (line["data"] != null) {
+                              content = Padding(
+                                padding: const EdgeInsets.only(bottom: 16),
+                                child: Row(
+                                  children: [
+                                    title,
+                                    Expanded(
+                                      flex: 2,
+                                      child: Text(
+                                        line["data"] as String,
+                                      ),
+                                    ),
+                                  ],
+                                ),
+                              );
+                            }
+
+                            if (line["list"] != null) {
+                              content = Row(
+                                children: [
+                                  title,
+                                  Expanded(
+                                    flex: 2,
+                                    child: Column(
+                                      children: [
+                                        ...line["list"].map(
+                                          (e) => Padding(
+                                            padding: const EdgeInsets.only(
+                                              bottom: 5,
+                                            ),
+                                            child: Ingredient(
+                                              name: '${e['name']}',
+                                              measure: '${e['measure']}',
+                                            ),
+                                          ),
+                                        )
+                                      ],
+                                    ),
+                                  ),
+                                ],
+                              );
+                            }
+                            final divider =
+                                lines.indexOf(line) != lines.length - 1
+                                    ? const Divider(
+                                        color: Color(AppColors.green),
+                                        thickness: 2,
+                                      )
+                                    : null;
+
+                            return ListTile(
+                              title: content,
+                              subtitle: divider,
+                            );
+                          },
+                        ),
+                      ],
+                    );
+                  },
+                  childCount: 1,
```

Comment:
all these widgets could be added to a new ConsumerStatefulWidget so that the tree looks less verbose and it is easier to maintain